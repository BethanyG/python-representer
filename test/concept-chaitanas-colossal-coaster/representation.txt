Module([FunctionDef('placeholder_0', arguments([], [arg('placeholder_1', lineno=4, col_offset=24, end_lineno=4, end_col_offset=37), arg('placeholder_2', lineno=4, col_offset=39, end_lineno=4, end_col_offset=51), arg('placeholder_3', lineno=4, col_offset=53, end_lineno=4, end_col_offset=64), arg('placeholder_4', lineno=4, col_offset=66, end_lineno=4, end_col_offset=77)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Assign([Name('placeholder_5', Store(), lineno=14, col_offset=4, end_lineno=14, end_col_offset=10)], IfExp(Compare(Name('placeholder_3', Load(), lineno=14, col_offset=30, end_lineno=14, end_col_offset=41), [Eq()], [Constant(1, lineno=14, col_offset=45, end_lineno=14, end_col_offset=46)], lineno=14, col_offset=30, end_lineno=14, end_col_offset=46), Name('placeholder_1', Load(), lineno=14, col_offset=13, end_lineno=14, end_col_offset=26), Name('placeholder_2', Load(), lineno=14, col_offset=52, end_lineno=14, end_col_offset=64), lineno=14, col_offset=13, end_lineno=14, end_col_offset=64), lineno=14, col_offset=4, end_lineno=14, end_col_offset=64), Expr(Call(Attribute(Name('placeholder_5', Load(), lineno=15, col_offset=4, end_lineno=15, end_col_offset=10), 'append', Load(), lineno=15, col_offset=4, end_lineno=15, end_col_offset=17), [Name('placeholder_4', Load(), lineno=15, col_offset=18, end_lineno=15, end_col_offset=29)], [], lineno=15, col_offset=4, end_lineno=15, end_col_offset=30), lineno=15, col_offset=4, end_lineno=15, end_col_offset=30), Return(Name('placeholder_5', Load(), lineno=16, col_offset=11, end_lineno=16, end_col_offset=17), lineno=16, col_offset=4, end_lineno=16, end_col_offset=17)], [], lineno=4, col_offset=0, end_lineno=16, end_col_offset=17), FunctionDef('placeholder_6', arguments([], [arg('placeholder_7', lineno=19, col_offset=19, end_lineno=19, end_col_offset=24), arg('placeholder_8', lineno=19, col_offset=26, end_lineno=19, end_col_offset=37)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Return(Call(Attribute(Name('placeholder_7', Load(), lineno=27, col_offset=11, end_lineno=27, end_col_offset=16), 'index', Load(), lineno=27, col_offset=11, end_lineno=27, end_col_offset=22), [Name('placeholder_8', Load(), lineno=27, col_offset=23, end_lineno=27, end_col_offset=34)], [], lineno=27, col_offset=11, end_lineno=27, end_col_offset=35), lineno=27, col_offset=4, end_lineno=27, end_col_offset=35)], [], lineno=19, col_offset=0, end_lineno=27, end_col_offset=35), FunctionDef('placeholder_9', arguments([], [arg('placeholder_7', lineno=30, col_offset=27, end_lineno=30, end_col_offset=32), arg('placeholder_10', lineno=30, col_offset=34, end_lineno=30, end_col_offset=39), arg('placeholder_4', lineno=30, col_offset=41, end_lineno=30, end_col_offset=52)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Expr(Call(Attribute(Name('placeholder_7', Load(), lineno=39, col_offset=4, end_lineno=39, end_col_offset=9), 'insert', Load(), lineno=39, col_offset=4, end_lineno=39, end_col_offset=16), [Name('placeholder_10', Load(), lineno=39, col_offset=17, end_lineno=39, end_col_offset=22), Name('placeholder_4', Load(), lineno=39, col_offset=24, end_lineno=39, end_col_offset=35)], [], lineno=39, col_offset=4, end_lineno=39, end_col_offset=36), lineno=39, col_offset=4, end_lineno=39, end_col_offset=36), Return(Name('placeholder_7', Load(), lineno=40, col_offset=11, end_lineno=40, end_col_offset=16), lineno=40, col_offset=4, end_lineno=40, end_col_offset=16)], [], lineno=30, col_offset=0, end_lineno=40, end_col_offset=16), FunctionDef('placeholder_11', arguments([], [arg('placeholder_7', lineno=43, col_offset=27, end_lineno=43, end_col_offset=32), arg('placeholder_4', lineno=43, col_offset=34, end_lineno=43, end_col_offset=45)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Expr(Call(Attribute(Name('placeholder_7', Load(), lineno=51, col_offset=4, end_lineno=51, end_col_offset=9), 'remove', Load(), lineno=51, col_offset=4, end_lineno=51, end_col_offset=16), [Name('placeholder_4', Load(), lineno=51, col_offset=17, end_lineno=51, end_col_offset=28)], [], lineno=51, col_offset=4, end_lineno=51, end_col_offset=29), lineno=51, col_offset=4, end_lineno=51, end_col_offset=29), Return(Name('placeholder_7', Load(), lineno=52, col_offset=11, end_lineno=52, end_col_offset=16), lineno=52, col_offset=4, end_lineno=52, end_col_offset=16)], [], lineno=43, col_offset=0, end_lineno=52, end_col_offset=16), FunctionDef('placeholder_12', arguments([], [arg('placeholder_7', lineno=55, col_offset=25, end_lineno=55, end_col_offset=30), arg('placeholder_4', lineno=55, col_offset=32, end_lineno=55, end_col_offset=43)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Return(Call(Attribute(Name('placeholder_7', Load(), lineno=63, col_offset=11, end_lineno=63, end_col_offset=16), 'count', Load(), lineno=63, col_offset=11, end_lineno=63, end_col_offset=22), [Name('placeholder_4', Load(), lineno=63, col_offset=23, end_lineno=63, end_col_offset=34)], [], lineno=63, col_offset=11, end_lineno=63, end_col_offset=35), lineno=63, col_offset=4, end_lineno=63, end_col_offset=35)], [], lineno=55, col_offset=0, end_lineno=63, end_col_offset=35), FunctionDef('placeholder_13', arguments([], [arg('placeholder_7', lineno=66, col_offset=27, end_lineno=66, end_col_offset=32)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Return(Call(Attribute(Name('placeholder_7', Load(), lineno=73, col_offset=11, end_lineno=73, end_col_offset=16), 'pop', Load(), lineno=73, col_offset=11, end_lineno=73, end_col_offset=20), [], [], lineno=73, col_offset=11, end_lineno=73, end_col_offset=22), lineno=73, col_offset=4, end_lineno=73, end_col_offset=22)], [], lineno=66, col_offset=0, end_lineno=73, end_col_offset=22), FunctionDef('placeholder_14', arguments([], [arg('placeholder_7', lineno=76, col_offset=17, end_lineno=76, end_col_offset=22)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Assign([Name('placeholder_15', Store(), lineno=83, col_offset=4, end_lineno=83, end_col_offset=13)], Subscript(Name('placeholder_7', Load(), lineno=83, col_offset=16, end_lineno=83, end_col_offset=21), Slice(lineno=83, col_offset=22, end_lineno=83, end_col_offset=23), Load(), lineno=83, col_offset=16, end_lineno=83, end_col_offset=24), lineno=83, col_offset=4, end_lineno=83, end_col_offset=24), Expr(Call(Attribute(Name('placeholder_15', Load(), lineno=84, col_offset=4, end_lineno=84, end_col_offset=13), 'sort', Load(), lineno=84, col_offset=4, end_lineno=84, end_col_offset=18), [], [], lineno=84, col_offset=4, end_lineno=84, end_col_offset=20), lineno=84, col_offset=4, end_lineno=84, end_col_offset=20), Return(Name('placeholder_15', Load(), lineno=85, col_offset=11, end_lineno=85, end_col_offset=20), lineno=85, col_offset=4, end_lineno=85, end_col_offset=20)], [], lineno=76, col_offset=0, end_lineno=85, end_col_offset=20)], [])