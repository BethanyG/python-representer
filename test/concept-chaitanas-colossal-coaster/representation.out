## BEGIN NORMALIZED CODE ##
def placeholder_0(placeholder_1, placeholder_2, placeholder_3, placeholder_4):
    placeholder_5 = placeholder_1 if placeholder_3 == 1 else placeholder_2
    placeholder_5.append(placeholder_4)
    return placeholder_5


def placeholder_6(placeholder_7, placeholder_8):
    return placeholder_7.index(placeholder_8)


def placeholder_9(placeholder_7, placeholder_10, placeholder_4):
    placeholder_7.insert(placeholder_10, placeholder_4)
    return placeholder_7


def placeholder_11(placeholder_7, placeholder_4):
    placeholder_7.remove(placeholder_4)
    return placeholder_7


def placeholder_12(placeholder_7, placeholder_4):
    return placeholder_7.count(placeholder_4)


def placeholder_13(placeholder_7):
    return placeholder_7.pop()


def placeholder_14(placeholder_7):
    placeholder_15 = placeholder_7[:]
    placeholder_15.sort()
    return placeholder_15

## END NORMALIZED CODE ##

## BEGIN TREE BEFORE ##
Module(
  [
    Expr(
      Constant(
        "Functions to manage and organize queues at Chaitana's roller coaster.",
        lineno=1,
        col_offset=0,
        end_lineno=1,
        end_col_offset=75),
      lineno=1,
      col_offset=0,
      end_lineno=1,
      end_col_offset=75),
    FunctionDef(
      'add_me_to_the_queue',
      arguments(
        [],
        [
          arg(
            'express_queue',
            lineno=4,
            col_offset=24,
            end_lineno=4,
            end_col_offset=37),
          arg(
            'normal_queue',
            lineno=4,
            col_offset=39,
            end_lineno=4,
            end_col_offset=51),
          arg(
            'ticket_type',
            lineno=4,
            col_offset=53,
            end_lineno=4,
            end_col_offset=64),
          arg(
            'person_name',
            lineno=4,
            col_offset=66,
            end_lineno=4,
            end_col_offset=77)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            "Add a person to the 'express' or 'normal' queue depending on the ticket number.\n\n    :param express_queue: list - names in the Fast-track queue.\n    :param normal_queue: list - names in the normal queue.\n    :param ticket_type: int - type of ticket. 1 = express, 0 = normal.\n    :param person_name: str - name of person to add to a queue.\n    :return: list - the (updated) queue the name was added to.\n    ",
            lineno=5,
            col_offset=4,
            end_lineno=12,
            end_col_offset=7),
          lineno=5,
          col_offset=4,
          end_lineno=12,
          end_col_offset=7),
        Assign(
          [
            Name(
              'result',
              Store(),
              lineno=14,
              col_offset=4,
              end_lineno=14,
              end_col_offset=10)],
          IfExp(
            Compare(
              Name(
                'ticket_type',
                Load(),
                lineno=14,
                col_offset=30,
                end_lineno=14,
                end_col_offset=41),
              [
                Eq()],
              [
                Constant(
                  1,
                  lineno=14,
                  col_offset=45,
                  end_lineno=14,
                  end_col_offset=46)],
              lineno=14,
              col_offset=30,
              end_lineno=14,
              end_col_offset=46),
            Name(
              'express_queue',
              Load(),
              lineno=14,
              col_offset=13,
              end_lineno=14,
              end_col_offset=26),
            Name(
              'normal_queue',
              Load(),
              lineno=14,
              col_offset=52,
              end_lineno=14,
              end_col_offset=64),
            lineno=14,
            col_offset=13,
            end_lineno=14,
            end_col_offset=64),
          lineno=14,
          col_offset=4,
          end_lineno=14,
          end_col_offset=64),
        Expr(
          Call(
            Attribute(
              Name(
                'result',
                Load(),
                lineno=15,
                col_offset=4,
                end_lineno=15,
                end_col_offset=10),
              'append',
              Load(),
              lineno=15,
              col_offset=4,
              end_lineno=15,
              end_col_offset=17),
            [
              Name(
                'person_name',
                Load(),
                lineno=15,
                col_offset=18,
                end_lineno=15,
                end_col_offset=29)],
            [],
            lineno=15,
            col_offset=4,
            end_lineno=15,
            end_col_offset=30),
          lineno=15,
          col_offset=4,
          end_lineno=15,
          end_col_offset=30),
        Return(
          Name(
            'result',
            Load(),
            lineno=16,
            col_offset=11,
            end_lineno=16,
            end_col_offset=17),
          lineno=16,
          col_offset=4,
          end_lineno=16,
          end_col_offset=17)],
      [],
      lineno=4,
      col_offset=0,
      end_lineno=16,
      end_col_offset=17),
    FunctionDef(
      'find_my_friend',
      arguments(
        [],
        [
          arg(
            'queue',
            lineno=19,
            col_offset=19,
            end_lineno=19,
            end_col_offset=24),
          arg(
            'friend_name',
            lineno=19,
            col_offset=26,
            end_lineno=19,
            end_col_offset=37)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            'Search the queue for a name and return their queue position (index).\n\n    :param queue: list - names in the queue.\n    :param friend_name: str - name of friend to find.\n    :return: int - index at which the friends name was found.\n    ',
            lineno=20,
            col_offset=4,
            end_lineno=25,
            end_col_offset=7),
          lineno=20,
          col_offset=4,
          end_lineno=25,
          end_col_offset=7),
        Return(
          Call(
            Attribute(
              Name(
                'queue',
                Load(),
                lineno=27,
                col_offset=11,
                end_lineno=27,
                end_col_offset=16),
              'index',
              Load(),
              lineno=27,
              col_offset=11,
              end_lineno=27,
              end_col_offset=22),
            [
              Name(
                'friend_name',
                Load(),
                lineno=27,
                col_offset=23,
                end_lineno=27,
                end_col_offset=34)],
            [],
            lineno=27,
            col_offset=11,
            end_lineno=27,
            end_col_offset=35),
          lineno=27,
          col_offset=4,
          end_lineno=27,
          end_col_offset=35)],
      [],
      lineno=19,
      col_offset=0,
      end_lineno=27,
      end_col_offset=35),
    FunctionDef(
      'add_me_with_my_friends',
      arguments(
        [],
        [
          arg(
            'queue',
            lineno=30,
            col_offset=27,
            end_lineno=30,
            end_col_offset=32),
          arg(
            'index',
            lineno=30,
            col_offset=34,
            end_lineno=30,
            end_col_offset=39),
          arg(
            'person_name',
            lineno=30,
            col_offset=41,
            end_lineno=30,
            end_col_offset=52)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            "Insert the late arrival's name at a specific index of the queue.\n\n    :param queue: list - names in the queue.\n    :param index: int - the index at which to add the new name.\n    :param person_name: str - the name to add.\n    :return: list - queue updated with new name.\n    ",
            lineno=31,
            col_offset=4,
            end_lineno=37,
            end_col_offset=7),
          lineno=31,
          col_offset=4,
          end_lineno=37,
          end_col_offset=7),
        Expr(
          Call(
            Attribute(
              Name(
                'queue',
                Load(),
                lineno=39,
                col_offset=4,
                end_lineno=39,
                end_col_offset=9),
              'insert',
              Load(),
              lineno=39,
              col_offset=4,
              end_lineno=39,
              end_col_offset=16),
            [
              Name(
                'index',
                Load(),
                lineno=39,
                col_offset=17,
                end_lineno=39,
                end_col_offset=22),
              Name(
                'person_name',
                Load(),
                lineno=39,
                col_offset=24,
                end_lineno=39,
                end_col_offset=35)],
            [],
            lineno=39,
            col_offset=4,
            end_lineno=39,
            end_col_offset=36),
          lineno=39,
          col_offset=4,
          end_lineno=39,
          end_col_offset=36),
        Return(
          Name(
            'queue',
            Load(),
            lineno=40,
            col_offset=11,
            end_lineno=40,
            end_col_offset=16),
          lineno=40,
          col_offset=4,
          end_lineno=40,
          end_col_offset=16)],
      [],
      lineno=30,
      col_offset=0,
      end_lineno=40,
      end_col_offset=16),
    FunctionDef(
      'remove_the_mean_person',
      arguments(
        [],
        [
          arg(
            'queue',
            lineno=43,
            col_offset=27,
            end_lineno=43,
            end_col_offset=32),
          arg(
            'person_name',
            lineno=43,
            col_offset=34,
            end_lineno=43,
            end_col_offset=45)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            'Remove the mean person from the queue by the provided name.\n\n    :param queue: list - names in the queue.\n    :param person_name: str - name of mean person.\n    :return: list - queue update with the mean persons name removed.\n    ',
            lineno=44,
            col_offset=4,
            end_lineno=49,
            end_col_offset=7),
          lineno=44,
          col_offset=4,
          end_lineno=49,
          end_col_offset=7),
        Expr(
          Call(
            Attribute(
              Name(
                'queue',
                Load(),
                lineno=51,
                col_offset=4,
                end_lineno=51,
                end_col_offset=9),
              'remove',
              Load(),
              lineno=51,
              col_offset=4,
              end_lineno=51,
              end_col_offset=16),
            [
              Name(
                'person_name',
                Load(),
                lineno=51,
                col_offset=17,
                end_lineno=51,
                end_col_offset=28)],
            [],
            lineno=51,
            col_offset=4,
            end_lineno=51,
            end_col_offset=29),
          lineno=51,
          col_offset=4,
          end_lineno=51,
          end_col_offset=29),
        Return(
          Name(
            'queue',
            Load(),
            lineno=52,
            col_offset=11,
            end_lineno=52,
            end_col_offset=16),
          lineno=52,
          col_offset=4,
          end_lineno=52,
          end_col_offset=16)],
      [],
      lineno=43,
      col_offset=0,
      end_lineno=52,
      end_col_offset=16),
    FunctionDef(
      'how_many_namefellows',
      arguments(
        [],
        [
          arg(
            'queue',
            lineno=55,
            col_offset=25,
            end_lineno=55,
            end_col_offset=30),
          arg(
            'person_name',
            lineno=55,
            col_offset=32,
            end_lineno=55,
            end_col_offset=43)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            'Count how many times the provided name appears in the queue.\n\n    :param queue: list - names in the queue.\n    :param person_name: str - name you wish to count or track.\n    :return: int - the number of times the name appears in the queue.\n    ',
            lineno=56,
            col_offset=4,
            end_lineno=61,
            end_col_offset=7),
          lineno=56,
          col_offset=4,
          end_lineno=61,
          end_col_offset=7),
        Return(
          Call(
            Attribute(
              Name(
                'queue',
                Load(),
                lineno=63,
                col_offset=11,
                end_lineno=63,
                end_col_offset=16),
              'count',
              Load(),
              lineno=63,
              col_offset=11,
              end_lineno=63,
              end_col_offset=22),
            [
              Name(
                'person_name',
                Load(),
                lineno=63,
                col_offset=23,
                end_lineno=63,
                end_col_offset=34)],
            [],
            lineno=63,
            col_offset=11,
            end_lineno=63,
            end_col_offset=35),
          lineno=63,
          col_offset=4,
          end_lineno=63,
          end_col_offset=35)],
      [],
      lineno=55,
      col_offset=0,
      end_lineno=63,
      end_col_offset=35),
    FunctionDef(
      'remove_the_last_person',
      arguments(
        [],
        [
          arg(
            'queue',
            lineno=66,
            col_offset=27,
            end_lineno=66,
            end_col_offset=32)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            'Remove the person in the last index from the queue and return their name.\n\n    :param queue: list - names in the queue.\n    :return: str - name that has been removed from the end of the queue.\n    ',
            lineno=67,
            col_offset=4,
            end_lineno=71,
            end_col_offset=7),
          lineno=67,
          col_offset=4,
          end_lineno=71,
          end_col_offset=7),
        Return(
          Call(
            Attribute(
              Name(
                'queue',
                Load(),
                lineno=73,
                col_offset=11,
                end_lineno=73,
                end_col_offset=16),
              'pop',
              Load(),
              lineno=73,
              col_offset=11,
              end_lineno=73,
              end_col_offset=20),
            [],
            [],
            lineno=73,
            col_offset=11,
            end_lineno=73,
            end_col_offset=22),
          lineno=73,
          col_offset=4,
          end_lineno=73,
          end_col_offset=22)],
      [],
      lineno=66,
      col_offset=0,
      end_lineno=73,
      end_col_offset=22),
    FunctionDef(
      'sorted_names',
      arguments(
        [],
        [
          arg(
            'queue',
            lineno=76,
            col_offset=17,
            end_lineno=76,
            end_col_offset=22)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            'Sort the names in the queue in alphabetical order and return the result.\n\n    :param queue: list - names in the queue.\n    :return: list - copy of the queue in alphabetical order.\n    ',
            lineno=77,
            col_offset=4,
            end_lineno=81,
            end_col_offset=7),
          lineno=77,
          col_offset=4,
          end_lineno=81,
          end_col_offset=7),
        Assign(
          [
            Name(
              'new_queue',
              Store(),
              lineno=83,
              col_offset=4,
              end_lineno=83,
              end_col_offset=13)],
          Subscript(
            Name(
              'queue',
              Load(),
              lineno=83,
              col_offset=16,
              end_lineno=83,
              end_col_offset=21),
            Slice(
              lineno=83,
              col_offset=22,
              end_lineno=83,
              end_col_offset=23),
            Load(),
            lineno=83,
            col_offset=16,
            end_lineno=83,
            end_col_offset=24),
          lineno=83,
          col_offset=4,
          end_lineno=83,
          end_col_offset=24),
        Expr(
          Call(
            Attribute(
              Name(
                'new_queue',
                Load(),
                lineno=84,
                col_offset=4,
                end_lineno=84,
                end_col_offset=13),
              'sort',
              Load(),
              lineno=84,
              col_offset=4,
              end_lineno=84,
              end_col_offset=18),
            [],
            [],
            lineno=84,
            col_offset=4,
            end_lineno=84,
            end_col_offset=20),
          lineno=84,
          col_offset=4,
          end_lineno=84,
          end_col_offset=20),
        Return(
          Name(
            'new_queue',
            Load(),
            lineno=85,
            col_offset=11,
            end_lineno=85,
            end_col_offset=20),
          lineno=85,
          col_offset=4,
          end_lineno=85,
          end_col_offset=20)],
      [],
      lineno=76,
      col_offset=0,
      end_lineno=85,
      end_col_offset=20)],
  [])
## END TREE BEFORE ##

## BEGIN NORMALIZED TREE ##
Module(
  [
    FunctionDef(
      'placeholder_0',
      arguments(
        [],
        [
          arg(
            'placeholder_1',
            lineno=4,
            col_offset=24,
            end_lineno=4,
            end_col_offset=37),
          arg(
            'placeholder_2',
            lineno=4,
            col_offset=39,
            end_lineno=4,
            end_col_offset=51),
          arg(
            'placeholder_3',
            lineno=4,
            col_offset=53,
            end_lineno=4,
            end_col_offset=64),
          arg(
            'placeholder_4',
            lineno=4,
            col_offset=66,
            end_lineno=4,
            end_col_offset=77)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Name(
              'placeholder_5',
              Store(),
              lineno=14,
              col_offset=4,
              end_lineno=14,
              end_col_offset=10)],
          IfExp(
            Compare(
              Name(
                'placeholder_3',
                Load(),
                lineno=14,
                col_offset=30,
                end_lineno=14,
                end_col_offset=41),
              [
                Eq()],
              [
                Constant(
                  1,
                  lineno=14,
                  col_offset=45,
                  end_lineno=14,
                  end_col_offset=46)],
              lineno=14,
              col_offset=30,
              end_lineno=14,
              end_col_offset=46),
            Name(
              'placeholder_1',
              Load(),
              lineno=14,
              col_offset=13,
              end_lineno=14,
              end_col_offset=26),
            Name(
              'placeholder_2',
              Load(),
              lineno=14,
              col_offset=52,
              end_lineno=14,
              end_col_offset=64),
            lineno=14,
            col_offset=13,
            end_lineno=14,
            end_col_offset=64),
          lineno=14,
          col_offset=4,
          end_lineno=14,
          end_col_offset=64),
        Expr(
          Call(
            Attribute(
              Name(
                'placeholder_5',
                Load(),
                lineno=15,
                col_offset=4,
                end_lineno=15,
                end_col_offset=10),
              'append',
              Load(),
              lineno=15,
              col_offset=4,
              end_lineno=15,
              end_col_offset=17),
            [
              Name(
                'placeholder_4',
                Load(),
                lineno=15,
                col_offset=18,
                end_lineno=15,
                end_col_offset=29)],
            [],
            lineno=15,
            col_offset=4,
            end_lineno=15,
            end_col_offset=30),
          lineno=15,
          col_offset=4,
          end_lineno=15,
          end_col_offset=30),
        Return(
          Name(
            'placeholder_5',
            Load(),
            lineno=16,
            col_offset=11,
            end_lineno=16,
            end_col_offset=17),
          lineno=16,
          col_offset=4,
          end_lineno=16,
          end_col_offset=17)],
      [],
      lineno=4,
      col_offset=0,
      end_lineno=16,
      end_col_offset=17),
    FunctionDef(
      'placeholder_6',
      arguments(
        [],
        [
          arg(
            'placeholder_7',
            lineno=19,
            col_offset=19,
            end_lineno=19,
            end_col_offset=24),
          arg(
            'placeholder_8',
            lineno=19,
            col_offset=26,
            end_lineno=19,
            end_col_offset=37)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Return(
          Call(
            Attribute(
              Name(
                'placeholder_7',
                Load(),
                lineno=27,
                col_offset=11,
                end_lineno=27,
                end_col_offset=16),
              'index',
              Load(),
              lineno=27,
              col_offset=11,
              end_lineno=27,
              end_col_offset=22),
            [
              Name(
                'placeholder_8',
                Load(),
                lineno=27,
                col_offset=23,
                end_lineno=27,
                end_col_offset=34)],
            [],
            lineno=27,
            col_offset=11,
            end_lineno=27,
            end_col_offset=35),
          lineno=27,
          col_offset=4,
          end_lineno=27,
          end_col_offset=35)],
      [],
      lineno=19,
      col_offset=0,
      end_lineno=27,
      end_col_offset=35),
    FunctionDef(
      'placeholder_9',
      arguments(
        [],
        [
          arg(
            'placeholder_7',
            lineno=30,
            col_offset=27,
            end_lineno=30,
            end_col_offset=32),
          arg(
            'placeholder_10',
            lineno=30,
            col_offset=34,
            end_lineno=30,
            end_col_offset=39),
          arg(
            'placeholder_4',
            lineno=30,
            col_offset=41,
            end_lineno=30,
            end_col_offset=52)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Call(
            Attribute(
              Name(
                'placeholder_7',
                Load(),
                lineno=39,
                col_offset=4,
                end_lineno=39,
                end_col_offset=9),
              'insert',
              Load(),
              lineno=39,
              col_offset=4,
              end_lineno=39,
              end_col_offset=16),
            [
              Name(
                'placeholder_10',
                Load(),
                lineno=39,
                col_offset=17,
                end_lineno=39,
                end_col_offset=22),
              Name(
                'placeholder_4',
                Load(),
                lineno=39,
                col_offset=24,
                end_lineno=39,
                end_col_offset=35)],
            [],
            lineno=39,
            col_offset=4,
            end_lineno=39,
            end_col_offset=36),
          lineno=39,
          col_offset=4,
          end_lineno=39,
          end_col_offset=36),
        Return(
          Name(
            'placeholder_7',
            Load(),
            lineno=40,
            col_offset=11,
            end_lineno=40,
            end_col_offset=16),
          lineno=40,
          col_offset=4,
          end_lineno=40,
          end_col_offset=16)],
      [],
      lineno=30,
      col_offset=0,
      end_lineno=40,
      end_col_offset=16),
    FunctionDef(
      'placeholder_11',
      arguments(
        [],
        [
          arg(
            'placeholder_7',
            lineno=43,
            col_offset=27,
            end_lineno=43,
            end_col_offset=32),
          arg(
            'placeholder_4',
            lineno=43,
            col_offset=34,
            end_lineno=43,
            end_col_offset=45)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Call(
            Attribute(
              Name(
                'placeholder_7',
                Load(),
                lineno=51,
                col_offset=4,
                end_lineno=51,
                end_col_offset=9),
              'remove',
              Load(),
              lineno=51,
              col_offset=4,
              end_lineno=51,
              end_col_offset=16),
            [
              Name(
                'placeholder_4',
                Load(),
                lineno=51,
                col_offset=17,
                end_lineno=51,
                end_col_offset=28)],
            [],
            lineno=51,
            col_offset=4,
            end_lineno=51,
            end_col_offset=29),
          lineno=51,
          col_offset=4,
          end_lineno=51,
          end_col_offset=29),
        Return(
          Name(
            'placeholder_7',
            Load(),
            lineno=52,
            col_offset=11,
            end_lineno=52,
            end_col_offset=16),
          lineno=52,
          col_offset=4,
          end_lineno=52,
          end_col_offset=16)],
      [],
      lineno=43,
      col_offset=0,
      end_lineno=52,
      end_col_offset=16),
    FunctionDef(
      'placeholder_12',
      arguments(
        [],
        [
          arg(
            'placeholder_7',
            lineno=55,
            col_offset=25,
            end_lineno=55,
            end_col_offset=30),
          arg(
            'placeholder_4',
            lineno=55,
            col_offset=32,
            end_lineno=55,
            end_col_offset=43)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Return(
          Call(
            Attribute(
              Name(
                'placeholder_7',
                Load(),
                lineno=63,
                col_offset=11,
                end_lineno=63,
                end_col_offset=16),
              'count',
              Load(),
              lineno=63,
              col_offset=11,
              end_lineno=63,
              end_col_offset=22),
            [
              Name(
                'placeholder_4',
                Load(),
                lineno=63,
                col_offset=23,
                end_lineno=63,
                end_col_offset=34)],
            [],
            lineno=63,
            col_offset=11,
            end_lineno=63,
            end_col_offset=35),
          lineno=63,
          col_offset=4,
          end_lineno=63,
          end_col_offset=35)],
      [],
      lineno=55,
      col_offset=0,
      end_lineno=63,
      end_col_offset=35),
    FunctionDef(
      'placeholder_13',
      arguments(
        [],
        [
          arg(
            'placeholder_7',
            lineno=66,
            col_offset=27,
            end_lineno=66,
            end_col_offset=32)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Return(
          Call(
            Attribute(
              Name(
                'placeholder_7',
                Load(),
                lineno=73,
                col_offset=11,
                end_lineno=73,
                end_col_offset=16),
              'pop',
              Load(),
              lineno=73,
              col_offset=11,
              end_lineno=73,
              end_col_offset=20),
            [],
            [],
            lineno=73,
            col_offset=11,
            end_lineno=73,
            end_col_offset=22),
          lineno=73,
          col_offset=4,
          end_lineno=73,
          end_col_offset=22)],
      [],
      lineno=66,
      col_offset=0,
      end_lineno=73,
      end_col_offset=22),
    FunctionDef(
      'placeholder_14',
      arguments(
        [],
        [
          arg(
            'placeholder_7',
            lineno=76,
            col_offset=17,
            end_lineno=76,
            end_col_offset=22)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Name(
              'placeholder_15',
              Store(),
              lineno=83,
              col_offset=4,
              end_lineno=83,
              end_col_offset=13)],
          Subscript(
            Name(
              'placeholder_7',
              Load(),
              lineno=83,
              col_offset=16,
              end_lineno=83,
              end_col_offset=21),
            Slice(
              lineno=83,
              col_offset=22,
              end_lineno=83,
              end_col_offset=23),
            Load(),
            lineno=83,
            col_offset=16,
            end_lineno=83,
            end_col_offset=24),
          lineno=83,
          col_offset=4,
          end_lineno=83,
          end_col_offset=24),
        Expr(
          Call(
            Attribute(
              Name(
                'placeholder_15',
                Load(),
                lineno=84,
                col_offset=4,
                end_lineno=84,
                end_col_offset=13),
              'sort',
              Load(),
              lineno=84,
              col_offset=4,
              end_lineno=84,
              end_col_offset=18),
            [],
            [],
            lineno=84,
            col_offset=4,
            end_lineno=84,
            end_col_offset=20),
          lineno=84,
          col_offset=4,
          end_lineno=84,
          end_col_offset=20),
        Return(
          Name(
            'placeholder_15',
            Load(),
            lineno=85,
            col_offset=11,
            end_lineno=85,
            end_col_offset=20),
          lineno=85,
          col_offset=4,
          end_lineno=85,
          end_col_offset=20)],
      [],
      lineno=76,
      col_offset=0,
      end_lineno=85,
      end_col_offset=20)],
  [])
## END NORMALIZED TREE ##